mesa-demos/README.md

```diff
@@ -0,0 +1,27 @@
+# Mesa-demos snap
+
+This project creates a working snap of the [mesa/demos](https://cgit.freedesktop.org/mesa/demos) repository. It includes many applications using the OpenGL, OpenGL|ES 1.1, OpenGL|ES 2.0, EGL APIs.
+
+The main aim is to use these demos as a tool to develop opengl, opengles and related interfaces in snapd.
+
+## Current state
+
+Working features:
+  - Currently the following applications have been enabled in the snap :
+      teapot
+      gears
+      tunnel2
+      es2info
+      es2tri
+      es2gears
+
+Known issues:
+  - Only validated with Intel (i965, i915) GPU drivers, so far.
+  - Only a few of the demos are included in the snap. Currently, each demo requires an env var to be set which is achieved through a wrapper. Having [lp:1583259](https://bugs.launchpad.net/snappy/+bug/1583259) would help when adding the others wholesale.
+  - 'strict' mode is not yet supported due to [lp:1611978](https://bugs.launchpad.net/snappy/+bug/1611978)
+
+TODO:
+ - Validate on post-Xenial
```

Comment:
I've tried to use an lxd container for Debian and Fedora but they failed for different reasons. I've added this task as a TO-DO to the readme and will get to it when I've completed my main task (which is implementing the opengles interface).